python/pyspark/pandas/series.py

```diff
@@ -6859,13 +6860,16 @@ def _reduce_for_stat_function(
         sfun : the stats function to be used for aggregation
         name : original pandas API name.
         axis : used only for sanity check because series only support index axis.
-        numeric_only : not used by this implementation, but passed down by stats functions
+        numeric_only : not used by this implementation, but passed down by stats functions.
         """
         axis = validate_axis(axis)
         if axis == 1:
             raise NotImplementedError("Series does not support columns axis.")
 
-        scol = sfun(self)
+        if not skipna and self.hasnans:
```

Comment:
`hasnans` is actually sort of expensive. We might need to cover this with https://github.com/apache/spark/blob/master/python/pyspark/pandas/config.py#L199 configuration.

Reply:
+1, see https://github.com/apache/spark/commit/e031d00b52e1973ea62f0a80da948f169c73dad9 as ref

Reply:
Makes sense! Modified.